### ISSUE TITLE
Support For Uninstallation of Apps via activity

### ISSUE THREAD

**Issue by Tharow** [MuntashirAkon/AppManager#1274]
### Please check before submitting an issue

- [X] I am using the latest version of App Manager
- [X] I have searched the issues and haven't found anything relevant
- [X] I have read the docs

### Describe a description of the new feature

AppManager already accepts INSTALL_PACKAGE action
It should accept UNINSTALL_PACKAGE action as well.

The idea behind this to allow AppManager to replace the built-in android PackageInstaller 
With AppManager supporting running as System, this seemed like the next step.

I'm available for any testing that my be required 

### Describe the solution you'd like

The AppManager to have an Uninstall Activity accepting UNINSTALL_PACKAGE action, without dependence on the built-in packageinstaller app.

### Additional context

activity requirements
- android.intent.action.DELETE
- android.intent.action.UNINSTALL_PACKAGE
- android.intent.category.DEFAULT
- data scheme: package

**Comment by MuntashirAkon** [comment]
> The idea behind this to allow AppManager to replace the built-in android PackageInstaller

App Manager cannot replace the built-in package installer because it (AM) only handles Android packages (APK). App Manager, for instance, does not handle APEX packages. The alternative idea is to remove the conflicting intent filters from App Manager itself in favour of another small user app that would redirect such requests to App Manager.

**Comment by Tharow** [comment]
I think you misunderstand The PackageInstaller.apk the system apk, not the service behind it
I want to be able to replace /system/priv-app/PackageInstaller/ (or it's google variant) with the app manager. and that apk seems to only handle the intents for the package installer service. it also handles the install dialog interface. 

It seemed like a minor change, and something i was surprised to find that the app manager didn't already intercept app deletion requests.

while it would be amazing to replace the entirety of the Package Installer with app-manager but we'd need to implement feature parity with it and keep it updated for it to include instant app installing apex handling etc. which i'm temped to look into but still.

**Comment by MuntashirAkon** [comment]
> while it would be amazing to replace the entirety of the Package Installer with app-manager but we'd need to implement feature parity with it and keep it updated for it to include instant app installing apex handling etc. which I'm temped to look into but still.

This is something no app can achieve even if they copy the entire codebase from the package installer. Because the app needs to support Android 5-14 whereas the package installer only supports the version it's packaged with. You need to understand the changes made from Android 5 to 14 and then reflect those changes in the app. I won't say that it's impossible, but it's impossible for _me_ to invest my time (which is very limited) on a feature like this.

**Comment by Tharow** [comment]
Yeah it sounded like hell from the get go. which is why i only wanted to be able to replace the front-end apk for the Package installer. not the actual service because that'd be insane (while i'm interested in finding out how i could get that to work with aosp thats a completely different thing). 

I just want to replace the front-end app with a different front-end app. with hopefully some extra features down the line

**Comment by MuntashirAkon** [comment]
> Yeah it sounded like hell from the get go. which is why i only wanted to be able to replace the front-end apk for the Package installer. not the actual service because that'd be insane (while i'm interested in finding out how i could get that to work with aosp thats a completely different thing).

I'm talking about the front-end APK. You can't replace Android built-in installer. The google installer is the same installer as the stock with renamed package name. Other installers (e.g., MIUI) uses a different skin to inject their own blobs/ads.

**Comment by MuntashirAkon** [comment]
> Oh, I See, I'm gonna try and make a universal version of the front end app. i doubt it will go well but it's worth a shot. the front end seemed quite empty when i looked through them so it should mostly just be compiling the changes between the versions. but that's probably ignorance talking. I'll give my self till the end of the year if i don't have it by then I'll close the issue.

This issue is originally about supporting app uninstallation, which is acceptable. Please create another issue, and I'll assign it to you.
